influxdb component config var formatting

nwesterhausen opened the pull request:
Updating the configuration variable section.

I wasn't sure that the customize variables should be included, since they apply to all components from my understanding.
```
- **component_config**, **component_config_domain**, **component_config_glob** (*Optional*): These attributes contains component-specific override values. See [Customizing devices and services](/getting-started/customizing-devices/) for format.
  - **override_measurement** (*Optional*): Measurement name to use for this component, takes precedence over the global 'override_measurement' and component-specific 'unit_of_measurement' attribute.
```

Related to #6385.

**Description:**


**Pull request in [home-assistant](https://github.com/home-assistant/home-assistant) (if applicable):** home-assistant/home-assistant#<home-assistant PR number goes here>

## Checklist:

- [ ] Branch: `next` is for changes and new documentation that will go public with the next [home-assistant](https://github.com/home-assistant/home-assistant) release. Fixes, changes and adjustments for the current release should be created against `current`.
- [ ] The documentation follow the [standards][standards].

[standards]: https://home-assistant.io/developers/documentation/standards/

nwesterhausen committed:
influxdb component config var formatting

Updating the configuration variable section.

I wasn't sure that the customize variables should be included, since they apply to all components from my understanding.

- **component_config**, **component_config_domain**, **component_config_glob** (*Optional*): These attributes contains component-specific override values. See [Customizing devices and services](/getting-started/customizing-devices/) for format.
  - **override_measurement** (*Optional*): Measurement name to use for this component, takes precedence over the global 'override_measurement' and component-specific 'unit_of_measurement' attribute.

Related to #6385.

nwesterhausen committed:
Fixed default values

based on the component influxdb.py

frenck commented on source/_components/influxdb.markdown:
```diff
@@ -25,28 +25,91 @@ influxdb:
+  required: false
+  default: localhost
+port:
+  type: int
```
`int` -> `integer`

frenck commented on source/_components/influxdb.markdown:
```diff
@@ -25,28 +25,91 @@ influxdb:
+  type: string
+  description: Name of the database to use. The database must already exist.
+  required: false
+  default: will use the default database of your influxdb instance
```
According to the code (maybe I misread it), the default is `home_assistant`.

https://github.com/home-assistant/home-assistant/blob/e903f7ffdafc7f5e0797f654c9b8314529412775/homeassistant/components/influxdb.py#L40

nwesterhausen replied:
Yes, I second guessed myself and misread the python. Thanks for catching that!

nwesterhausen commented:
Right now, this component page links to the blog post on influxdb (from 2015) which outlines getting influxdb set up and using it with home assistant. The process in the blog post will still work, it uses an old version of influxdb. The current version of influxdb no longer has [built-in administration](https://docs.influxdata.com/influxdb/v1.6/about_the_project/releasenotes-changelog/#deprecations-1). 

Do we want to have updated instructions or possibly link to [this forum post](https://community.home-assistant.io/t/complete-guide-on-setting-up-grafana-influxdb-with-home-assistant-using-official-docker-images/42860) as an up to date alternative for the blog post?

frenck reviewed (changes requested):
@nwesterhausen Thank you for doing this! 👍 

About the customize variables, I would opt to add them, each individually (Rather be explicit on the options).

I'm not for updating/rewriting blog posts. Those are a snapshot of history.
Maybe move the update of the contents to a secondary PR?

I also left some comments on the PR itself. Could you please take a look?

nwesterhausen committed:
Update influxdb.markdown

Fix both 'int' -> 'integer'

Fix default database to correct value of 'home_assistant'

nwesterhausen commented:
I agree we shouldn't change the blog post. I should have been more explicit in my suggestion: either putting the instructions into this page, a secondary page with more instructions on influx db, or link to the forum post which has more up to date instructions.

I think based off some other components which involve extra steps, it makes sense to put the instructions into this component page.

frenck commented:
@nwesterhausen Agreed.

nwesterhausen committed:
Update influxdb.markdown

Added in component customization values to config variables. Left with default description from original component page. I'm checking to see if there is a better way to describe these.

nwesterhausen committed:
Update influxdb.markdown

Fixed wrong type for override_measurement.

Happy with description of custom values.

nwesterhausen commented:
I initially misunderstood the custom_* variables and what they would be used for. It turns out it actually is really simple: if there is a set of entities or a domain that you want to specify a special measurement value for the database (e.g. "% Humidity" instead of just "%" for humidity values) you could add a configuration value here that would accomplish that for you.

nwesterhausen commented:
I believe this is ready to be merged.

nwesterhausen committed:
Required Changes

Fixed grammar for the three customize variables.

frenck commented:
Thank, @nwesterhausen! 👍 

Happy Hacktoberfest 🎉 

../Frenck
